open-craft/frontend-app-course-authoring
# feat: Avoid close Tags drawer on edit

## Description

- To avoid lose user data
- Wrap with ContentTagsDrawerSheet to build drawer on MFE

## Supporting information

- Github issue: https://github.com/openedx/modular-learning/issues/209
- This PR doesn't include:
> The footer of the drawer animates to draw the user's attention to the actions (see [Figma](https://www.figma.com/file/DmUC2e7BAoQsGDq4KgApPa/Content-Tagging-MVP?type=design&node-id=13585-6449&mode=design&t=hgTeNriivLM74TtO-4)). Perhaps like [Blink 2 at this link](https://webcode.tools/css-generator/keyframe-animation)
- Internal ticket: [FAL-3711](https://tasks.opencraft.com/browse/FAL-3711)
- Depends on: https://github.com/openedx/frontend-app-course-authoring/pull/939

## Testing instructions

- Use this branch of edx-platform: https://github.com/openedx/edx-platform/pull/34490
- Run `make requirements` on cms shell.
- Make sure you have some sample taxonomy/tags data setup: https://github.com/open-craft/taxonomy-sample-data
- Go into a course and click on "Manage tags" option for any unit/block to open the tags drawer
- Click on `Edit tags`
- Delete a tags and verify that can't close drawer when clicking outside.
- Cancel and verify that you can close drawer when clicking outside.
- Click on `Edit tags`
- Add new tags and verify that can't close drawer when clicking outside.
